### DISCUSSION TITLE
Timeout sur la récupération des infos

### DISCUSSION THREAD

**Discussion by OUARZA** [OUARZA/Propluvia#40] [Ideas]
Possible de rallonger le timeout ?

![image](https://github.com/OUARZA/Propluvia/assets/34892335/d0d1c6a8-3417-464c-8b63-84b51ca30781)

**Comment by elmerdu69** [comment]
je vais regarder comment allonger mais c'est typiquement une erreur quand l'API ne répond pas en face.

Si pas possible peut être qu'il faudrait mettre en place un mécanisme pour relancer l'appel quelques minutes plus tard

**Reply by OUARZA** [reply]
Je pense en fait que c'est pas un problème de non réponse.
Je pense que c'est plutôt que les données sont lourdes à afficher, non ?

**Reply by elmerdu69** [reply]
Peut être que tu as raison. Est-ce que tu vois souvent cette erreur ? de mon coté quasiment jamais.

**Reply by OUARZA** [reply]
Alors moi non, mais la personne du forum oui.
je viens de tester avec son code postal 66065, regarde
![image](https://github.com/OUARZA/Propluvia/assets/34892335/e59ee15f-f5c6-4ccf-9a5c-50d2ab59c805)

23096 c'est un code postal qui est en alerte crise aussi

**Reply by elmerdu69** [reply]
ok, mes essais de cet aprèm me font dire que le timeout est lié (comme tu le précisais) à la taille du fichier json mais aussi à la bande passante de la connexion internet.
Je sais comment modifier ce timeout mais il faut reprendre toute la partie cURL. Je m'y mets demain

**Reply by elmerdu69** [reply]
modification effectuée, un nouveau PR à valider
j'ai fixé le timeout de récupération à 15s, ça devrait être large. 
J'ai effectué quelques test mais pas suffisamment sur tous les cas. Peux tu en effectuer de ton coté?

**Comment by OUARZA** [comment]
Hello,
Je vais mettre en béta et je vais demander à la personne du forum de tester.
Mathieu

**Comment by OUARZA** [comment]
#45
